### PR TITLE
appium-inspector: init at 2024.3.1

### PR DESCRIPTION
**Pull request by NyCodeGHG**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ap/appium-inspector/package.nix [hunk 1]
```diff
@@ -0,0 +1,61 @@
+{ fetchFromGitHub
+, buildNpmPackage
+, makeWrapper
+, electron_13
+, makeDesktopItem
+, stdenv
+, lib
+}:
+let
+  electron = electron_13;
```
Comment (SuperSandro2000) [inline pkgs/by-name/ap/appium-inspector/package.nix:10]: This is very old, can't we use something newer?

File: pkgs/by-name/ap/appium-inspector/package.nix [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    cp -r './node_modules' "$out/share/appium-inspector/node_modules"
+    cp -r './dist' "$out/share/appium-inspector/"
+    cp './package.json' "$out/share/appium-inspector/package.json"
+
+    # executable wrapper
+    makeWrapper '${electron}/bin/electron' "$out/bin/appium-inspector" \
+      --add-flags "$out/share/appium-inspector/dist/main.js" \
+      --set NODE_ENV production
+
+    ln -s "${desktopItem}/share/applications" "$out/share/applications"
```
Comment (SuperSandro2000) [inline pkgs/by-name/ap/appium-inspector/package.nix:42]: Please use copyDesktopItems instead

File: pkgs/by-name/ap/appium-inspector/package.nix (hunk 1 above)
Comment (NyCodeGHG) [reply pkgs/by-name/ap/appium-inspector/package.nix:10]: I also mentioned this to the upstream maintainers, it doesn't work on anything newer unfortunately 
see https://github.com/appium/appium-inspector/issues/1321#issuecomment-1926911425

File: pkgs/by-name/ap/appium-inspector/package.nix (hunk 2 above)
Comment (NyCodeGHG) [reply pkgs/by-name/ap/appium-inspector/package.nix]: done!

**Comment by mweinelt** [comment]
We removed EOL electron versions two days ago in https://github.com/NixOS/nixpkgs/pull/296152, so merging this caused an evaluation error.

Was reverted in #298746 and you'll have to think how to go about reintroducing it.

**Comment by NyCodeGHG** [comment]
thank you for telling me, I probably won't reintroduce this, as it seems upstream doesn't care about the EOL version

**Pull request merged**